src/rust/en.mangairo/src/parser.rs

```diff
@@ -0,0 +1,354 @@
+use aidoku::{
+	error::Result, helpers::uri::QueryParameters, prelude::*, std::html::Node, std::String,
+	std::Vec, Chapter, Filter, FilterType, Manga, MangaContentRating, MangaStatus, MangaViewer,
+	Page,
+};
+extern crate alloc;
+use alloc::string::ToString;
+
+pub const BASE_URL: &str = "https://w.mangairo.com";
+pub const USER_AGENT: &str = "Mozilla/5.0 (Macintosh; Intel Mac OS X 13_3_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/113.0.0.0 Safari/537.36";
+
+pub fn parse_manga_list(html: Node, page: i32) -> (Vec<Manga>, bool) {
+	let mut result: Vec<Manga> = Vec::new();
+	for page in html.select(".story-item").array() {
+		let obj = page.as_node().expect("node array");
+
+		let id = obj.select(".story-name a").attr("href").read();
+		let title = obj.select(".story-name a ").text().read();
+		let cover = obj.select(".story-list-img img").attr("src").read();
+
+		if !id.is_empty() && !title.is_empty() && !cover.is_empty() {
+			result.push(Manga {
+				id,
+				cover,
+				title,
+				..Default::default()
+			});
+		}
+	}
+
+	// Example: 'Total: 38,202 stories'
+	let mut total_str = html.select(".quantitychapter").text().read();
+	total_str = total_str.replace("Total: ", "");
+	total_str = total_str.replace(" stories", "");
+	total_str = total_str.chars().filter(|&c| c != ',').collect();
+
+	let has_more = total_str
+		.parse::<i32>()
+		.map_or(false, |value| value > result.len() as i32 * page);
+	(result, has_more)
+}
+
+pub fn parse_manga_details(html: Node, id: String) -> Result<Manga> {
+	let title = html
+		.select(".breadcrumbs p span a span")
+		.last()
+		.text()
+		.read();
+	let cover = html.select(".avatar").attr("src").read();
+	let description = html
+		.select("div#story_discription p")
+		.text()
+		.read()
+		.trim()
+		.to_string();
+	let status_str = html
+		.select(".story_info_right li:nth-child(5) a")
+		.text()
+		.read()
+		.to_lowercase();
+
+	let url = format!("{}", &id);
+
+	let mut authors: Vec<String> = Vec::new();
+	html.select(".story_info_right li:nth-child(3) a")
+		.array()
+		.for_each(|tag| {
+			authors.push(String::from(
+				tag.as_node().expect("node array").text().read().trim(),
+			))
+		});
+	let author = authors.join(", ");
```

Comment:
I think you could do this
```suggestion
	let author = html
		.select(".story_info_right li:nth-child(3) a")
		.array()
		.map(|tag| {
			String::from(
				tag.as_node().expect("node array").text().read().trim(),
			)
		})
		.join(", ");
```